File: digiLoLAPI/src/main/kotlin/com/hjkwon93/digiLoLAPI/common/ChampionIdMap.kt
package com.hjkwon93.digiLoLAPI.common

import com.google.gson.JsonParser
import java.net.HttpURLConnection
import java.net.URL

object ChampionIdMap {
 val map = mutableMapOf<Int,String>()

  init {
    val url = URL("https://ddragon.leagueoflegends.com/cdn/10.16.1/data/en_US/champion.json")
    with(url.openConnection() as HttpURLConnection){
      requestMethod = "GET"
      val jsonObj = JsonParser.parseString(inputStream.bufferedReader().readText()).asJsonObject
      val data = jsonObj.get("data").asJsonObject
      val keys = data.keySet()
      keys.forEach{champName->
        run {
          val champId = data.get(champName).asJsonObject.get("key").asInt
          map.put(champId,champName.toString())
        }
      }
    }
  }
}

File: DigiLoLAPI_deno/common/ChampionIdMap.ts
import axiod from "https://deno.land/x/axiod/mod.ts";
import config from "../config.ts";
import fixedEncodeURI from "./fixedEncodeURI.ts";
class ChampionIdMap {
  private map: Map<number, string>;
  constructor() {
    this.map = new Map<number, string>();
  }
  async init() {
    try {
      const res = await axiod.get(
        fixedEncodeURI(
          `https://ddragon.leagueoflegends.com/cdn/${config.gameVersion}/data/en_US/champion.json`,
        ),
      );
      const championData = res.data.data;
      for (const key in championData) {
        this.map.set(parseInt(championData[key].key), key.toString());
      }
      return true;
    } catch (error) {
      throw error;
    }
  }
  getChampionName(chapionId: number) {
    return this.map.get(chapionId);
  }
}

export default new ChampionIdMap();
